### PR TITLE
graph/layout: add Isomap layout

### PR DESCRIPTION
**Pull request by kortschak**
Isomap is not a particularly good layout — it doesn't handle graphs with more than one connected component, doesn't scale well, and cannot prevent overlap between indistinguishably distant nodes — but it is a single iteration update function and can be used in some places (for example where node similarity/indistinguishability is interesting), and the R³ layout is just as expensive code wise and computation wise as the R² case.

Also, it's cool how easy it is to write.

Please take a look.

### CODE REVIEW

File: graph/layout/isomap.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+)
+
+// IsomapR2 implements a graph layout algorithm based on the Isomap
+// non-linear dimensionality reduction method. Coordinates of nodes
+// are computed by finding a Torgerson multidimensional scaling of
+// the shortest path distances between all pairs of node in the graph.
+// The all pair shortest path distances are calculated using the
+// Floyd-Warshall algorithm and so IsomapR2 will not scale to large
+// graphs. Graphs with more than one connected component cannot be
+// layed out by IsomapR2.
```
[the remark this answers is not in the document]
Comment (kortschak) [reply graph/layout/isomap.go]: I'd like to think so. The First sentence should probably start with "The all pairs shortest path ...". Is there a  particular bit that is grating other than that?

[the remark this answers is not in the document]
Comment (vladimir-ch) [reply graph/layout/isomap.go]: Sorry, I should have been more specific. I'm referring to "layed out" https://brians.wsu.edu/2016/05/19/layed-laid/

[the remark this answers is not in the document]
Comment (kortschak) [reply graph/layout/isomap.go]: Thanks. Fixed.

**Comment by kortschak** [comment]
GitHub just failed to squash, then asked to try again. It then did a merge commit.

**Comment by vladimir-ch** [comment]
Maybe some kind of race condition because I was rebasing and merging the Dlatbs branch around the same time.

**Comment by kortschak** [comment]
That should not have caused that failure, and github should have failed out rather than changing merge strategy.

**Pull request merged**